src/Stratis.Bitcoin/Consensus/BestChainIndexer.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Consensus
+{
+    /// <summary>
+    /// An indexer of the consensus chain.
+    /// </summary>
+    public class BestChainIndexer
+    {
+        private readonly IConsensusManager consensusManager;
+        private readonly Dictionary<int, ChainedHeader> blocksByHeight = new Dictionary<int, ChainedHeader>();
+        private readonly object lockObj = new object();
+
+        internal BestChainIndexer(IConsensusManager consensusManager)
+        {
+            this.consensusManager = consensusManager;
+        }
+
+        public ChainedHeader Tip => this.consensusManager.Tip;
```

Comment:
if you say 
```
Usage will be as following this.ConsensusManager.BestChainIndexer .GetBlock(1)
```

then there is no need to expose tip, user will always be able to access it via CM.

Reply:
Yeah but ideally overtime only the Indexer can be used, also this is to replace all CC methods if we find we don't need to use it we can delete it.

Reply:
`Yeah but ideally overtime only the Indexer can be used`

In that case it shouldn't be accessible through CM but through CI